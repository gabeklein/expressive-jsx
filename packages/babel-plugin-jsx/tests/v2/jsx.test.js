describe("implicit return", () => {
  transform("arrow function", `
    () => {
      <this>Hello World</this>
    }
  `)

  transform("with statements", `
    () => {
      const [x, setX] = useState();
      
      useEffect(() => {
        void x;
        setX();
      }, []);

      <div />
    }
  `)

  transform("function declaration", `
    function App() {
      <this>Hello World</this>
    }
  `)

  transform("function expression", `
    const App = function() {
      <this>Hello World</this>
    }
  `)
});

describe("children", () => {
  transform("fragment", `
    () => {
      <>
        <div>Foo</div>
        <div>Bar</div>
      </>
    }
  `)

  transform("multiple", `
    () => {
      <this>
        <div>Foo</div>
        <div>Bar</div>
      </this>
    }
  `)
})

describe("MemberExpression tags", () => {
  transform("will passthru", `
    () => {
      <Foo.Bar />
    }
  `);

  transform("will apply modifiers", `
    () => {
      Bar: {
        color: red;
      }
      
      <Foo.Bar />
    }
  `)
})