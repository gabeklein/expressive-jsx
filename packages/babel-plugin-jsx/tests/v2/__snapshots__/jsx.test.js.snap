// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberExpression tags will apply modifiers 1`] = `
  () => <Foo.Bar className="Bar_6b5" />;
  css(\`
  	.Bar_6b5 {
  		color: red;
  	}
  \`);
`;

exports[`MemberExpression tags will passthru 1`] = `  () => <Foo.Bar />;`;

exports[`children fragment 1`] = `
  () => (
    <>
      <div>Foo</div>
      <div>Bar</div>
    </>
  );
`;

exports[`children multiple 1`] = `
  () => (
    <div>
      <div>Foo</div>
      <div>Bar</div>
    </div>
  );
`;

exports[`implicit return arrow function 1`] = `  () => <div>Hello World</div>;`;

exports[`implicit return function declaration 1`] = `
  function App() {
    return <div>Hello World</div>;
  }
`;

exports[`implicit return function expression 1`] = `
  const App = function () {
    return <div>Hello World</div>;
  };
`;

exports[`implicit return with statements 1`] = `
  () => {
    const [x, setX] = useState();
    useEffect(() => {
      void x;
      setX();
    }, []);
  
    return <div />;
  };
`;
